### DISCUSSION TITLE
Is there interest to use an interactive tutorial engine I am working on

### DISCUSSION THREAD

**Discussion by hrgdavor** [jsxcad/JSxCAD#1156] [Ideas]
I would like you to look at 2 instances of my attempts at making interactive tutorial 
 - has part with explanations and snippets
 - uses markdown file as source for the content
 - shows and executes code examples
 - allows for changing examples to play with code

I am wondering if you would be interested in using that for your documentation notebooks.
Or you may find parts of my code useful as I plan to separate useful parts for easier reuse (runner, code editor).

my first prototype of interactive tutorial engine is here:
https://3d.hrg.hr/jscad/three/loop/loop.html#1
and source for the content is read from this md file is here
https://3d.hrg.hr/jscad/three/loop/loop.md

latest version of my interactive tutorial engine is here where I made a tutorial about JSX
https://hrgdavor.github.io/jsx6/demistify/
and source for the content is read from this md file is here
https://hrgdavor.github.io/jsx6/demistify/demistify.jsx.md

**Comment by pentacular** [comment]
Perhaps -- I'd like to see how your system evolves.

The approach I've taken so far is to have the notebook be a dialect of es6 javascript, with support for easy embedding and generation of markdown.

Coupled with a persistent editing system, capable of distributed code evaluation.

This is the result of a lot of evolution, so the code needs a lot of cleaning up.

At this point, I think it's mostly converged toward something useful, but I'm always interested in replacing things at the component level.

I'll be interested in seeing how what you develop turns out. :)
